File: job-portal-backend/jobservice/src/main/java/com/carrerconnect/jobservice/service/JobService.java
package com.carrerconnect.jobservice.service;

import java.util.List;

import com.carrerconnect.jobservice.dto.ApplicationDTO;
import com.carrerconnect.jobservice.dto.JobDTO;
import com.carrerconnect.jobservice.exception.CarrerConnectException;

public interface JobService {
    JobDTO createJob(JobDTO jobDTO);
    JobDTO getJobById(Long id) throws CarrerConnectException;
    List<JobDTO> getAllJobs();
    List<JobDTO> getJobsByUserId(Long userId) throws CarrerConnectException;
    JobDTO updateJob(Long id, JobDTO jobDTO);
    void deleteJob(Long id) throws CarrerConnectException;
    
    // Add this method for job applications
    List<ApplicationDTO> getApplicationsForJob(Long jobId);
}

File: job-portal-frontend/src/api/jobService.js
import apiClient from './api';

export const getAllJobs = () => {
    return apiClient.get('/jobs');
};

export const getJobById = (id) => {
    return apiClient.get(`/jobs/${id}`);
};

export const createJob = (jobData) => {
    return apiClient.post('/jobs', jobData);
};

export const updateJob = (id, jobData) => {
    return apiClient.put(`/jobs/${id}`, jobData);
};

export const deleteJob = (id) => {
    return apiClient.delete(`/jobs/${id}`);
};

export const getJobsByUserId = (userId) => {
    return apiClient.get(`/jobs/user/${userId}`);
};